### DISCUSSION TITLE
What enhanced benefit does unstable_data offer over the standard alternatives?

### DISCUSSION THREAD

**Discussion by bluefire2121** [remix-run/remix#9883] [Q&A] [answered]
Forgive me if there's a misunderstanding, but is `unstable_data` redundant? Can one not just modify the response object directly in a loader/action and throw the response afterwards, or even create a new response to return?

What is the superior benefit of using `unstable_data`?

Example:
```tsx
response.status = 418;
response.headers.set("Catche-Control", "no-store");
throw response;
```

Another Example:
```tsx
return new Response(null, {
  status: 303,
  headers: {
    Location: "/some/where/else",
  },
});
```

**Comment by sergiodxa** [comment, marked as answer]
The response object is not received anymore with single fetch

If you return a custom response object you will bail out of single fetch

The unstable_data is used by Remix to allow single fetch and turbo-stream responses

**Reply by bluefire2121** [reply]
Thank you. This makes sense. For single fetch, unstable_data isn't optional.

One additional question for you @sergiodxa, how do you `redirect` to a location in an action using the `unstable_data` method?

Following the [current documentation](https://remix.run/docs/en/main/utils/data), it should just be 

```tsx
export const action = unstable_defineAction( async ({ params }) => {

return data(
null, 
{
status: 302,
headers: {
Location: "/go/some/where"
}
});

});
```

any reason why the above wouldn't work?

**Reply by bluefire2121** [reply]
In case anyone finds this in the future, just use the ```redirect()``` function. ```json()``` and ```defer()``` are gone, but ```redirect()``` is hanging around.
